### PR TITLE
Issue #12397 - Add more default excludes to GzipHandler.

### LINKED ISSUE #12397: `.tgz` files are double-gzipped
### Jetty version(s)

12.0.14

### Jetty Environment

EE 9
EE 10

### Java version/vendor

```
openjdk version "17.0.12" 2024-07-16
OpenJDK Runtime Environment (build 17.0.12+7-Ubuntu-1ubuntu222.04)
OpenJDK 64-Bit Server VM (build 17.0.12+7-Ubuntu-1ubuntu222.04, mixed mode, sharing)
```

### OS type/version

Ubuntu 22.04.5 LTS x86_64

### Context

See [JENKINS-73942](https://issues.jenkins.io/browse/JENKINS-73942).

### Steps to reproduce

```bash
$ wget https://repo1.maven.org/maven2/org/eclipse/jetty/jetty-home/12.0.14/jetty-home-12.0.14.tar.gz
$ tar -xzvf jetty-home-12.0.14.tar.gz
$ export JETTY_HOME=/path/to/jetty-home-12.0.14
$ git clone https://github.com/basil/jetty-tgz-repro.git
$ export JETTY_BASE=/path/to/jetty-tgz-repro
```

Observe the files are identical:

```bash
$ sha256sum $JETTY_BASE/webapps/static/resources/example.tar.gz $JETTY_BASE/webapps/static/resources/example.tgz
45749065add6664e1fc7c5a75665dd9dac85f70a1444612b895b222538601a13  webapps/static/resources/example.tar.gz
45749065add6664e1fc7c5a75665dd9dac85f70a1444612b895b222538601a13  webapps/static/resources/example.tgz
```

Start the server:

```bash
$ $JETTY_HOME/bin/jetty.sh start
```

Visit http://127.0.0.1:8080/static/ in Firefox and download both files.

### Expected results

The files should be identical.

### Actual results

The files are not identical, as the `.tgz` file is (erroneously!) double-gzipped. The `.tar.gz` file is correct.

```
$ sha256sum example.tar.gz example.tgz 
45749065add6664e1fc7c5a75665dd9dac85f70a1444612b895b222538601a13  example.tar.gz
1ca0c922947f4426b90526f3ec5427d2123b4b6fec92e66f8b14cec792271dae  example.tgz
```

### Note

The problem only appears when the `gzip` module is enabled.

The problem disappeared after I unzipped `$JETTY_HOME/lib/jetty-http-12.0.14.jar`, updated `org/eclipse/jetty/http/mime.properties` to replace `tgz=application/x-gtar` with `tgz=application/gzip`, re-packed the JAR, and relaunched Jetty.

### PR DESCRIPTION
**Pull request by joakime**
+ Exclude more compressed mime-types
+ Exclude more compressed path extensions
+ Add missing compressed file types to mime.properties

Fixes: #12397

### CODE REVIEW

**Comment by joakime** [comment]
@gregw with this change, I see yet another thing that MimeTypes should probably do (flag if a mime-type represents compressed content that shouldn't be recompressed).

What do you think about refactoring the MimeTypes configuration files a great deal?

We could use json, and our jetty-util-ajax lib ...

``` json
[
  "application/gzip" : {
    "extensions": [ ".gz", ".gzip" ],
    "assumedCharset": "",
    "inferredCharset": ""
    "ignoreCharset": "true"
    "compressedType": "true"
  },
  "application/json" : {
    "extensions": [ ".json" ],
    "assumedCharset": "utf-8",
    "inferredCharset": ""
    "ignoreCharset": "false"
    "compressedType": "false"
  }
]
```

The idea is that this becomes our preferred configuration file, but we still allow the use of the older `mime.properties` and `encoding.properties` as an alternate deprecated configuration technique.

File: jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/gzip/GzipHandler.java [hunk 1]
```diff
@@ -85,13 +85,27 @@ else if (type.startsWith("image/") ||
                 type.startsWith("video/"))
                 _mimeTypes.exclude(type);
         }
+
         _mimeTypes.exclude("application/compress");
+        _paths.exclude("*.z");
```
Comment (janbartel) [inline jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/gzip/GzipHandler.java:90]: That seems a very broad brush stroke. Also I don't see this mimetype listed on https://www.iana.org/assignments/media-types/media-types.txt

**Review by janbartel — CHANGES_REQUESTED** [review]

File: jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/gzip/GzipHandler.java (hunk 1 above)
Comment (joakime) [reply jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/gzip/GzipHandler.java:90]: This was taken from our mime.properties file.

Comment (janbartel) [reply jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/gzip/GzipHandler.java:90]: yikes

**Review by janbartel — APPROVED** [review]

**Pull request merged**